apps/api/src/domain/models/playlist/commands/add-audio-item-to-playlist/add-audio-item-to-playlist.command.integration.spec.ts

```diff
@@ -0,0 +1,235 @@
+
+    let commandAssertionDependencies: CommandAssertionDependencies;
+
+    beforeAll(async () => {
+        ({ testRepositoryProvider, commandHandlerService, idManager, app } =
+            await setUpIntegrationTest({
+                ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+            }).catch((error) => {
+                throw error;
+            }));
+
+        commandAssertionDependencies = {
+            testRepositoryProvider,
+            idManager,
+            commandHandlerService,
+        };
+    });
+
+    afterAll(async () => {
+        await app.close();
+    });
+
+    beforeEach(async () => {
+        await testRepositoryProvider.testSetup();
+    });
+
+    describe('when the command is valid', () => {
+        it('should succeed', async () => {
+            await assertCommandSuccess(commandAssertionDependencies, {
+                systemUserId: dummySystemUserId,
+                buildValidCommandFSA,
+                initialState: new DeluxeInMemoryStore({
+                    [AggregateType.playlist]: [existingPlaylist],
+                    [AggregateType.audioItem]: [existingAudioItem],
+                }).fetchFullSnapshotInLegacyFormat(),
+                checkStateOnSuccess: async ({
+                    aggregateCompositeIdentifier: { id: playlistId },
+                    audioItemId,
+                }: AddAudioItemToPlaylist) => {
+                    const audioSearchResult = await testRepositoryProvider
+                        .forResource(ResourceType.audioItem)
+                        .fetchById(audioItemId);
+
+                    expect(audioSearchResult).not.toBe(NotFound);
+
+                    const playlistSearchResult = await testRepositoryProvider
+                        .forResource(ResourceType.playlist)
+                        .fetchById(playlistId);
+
+                    expect(playlistSearchResult).not.toBe(NotFound);
+
+                    const playlist = playlistSearchResult as Playlist;
+
+                    const isAudioItemInPlaylist = playlist.has({
+                        type: AggregateType.audioItem,
+                        id: audioItemId,
+                    });
+
+                    expect(isAudioItemInPlaylist).toBe(true);
+
+                    assertEventRecordPersisted(
+                        playlist,
+                        'AUDIO_ITEM_ADDED_TO_PLAYLIST',
+                        dummySystemUserId
+                    );
+                },
+            });
+        });
+    });
+
+    describe('when the command is invalid', () => {
+        describe('when the audio item is already in the playlist', () => {
+            it('should fail with the expected errors', async () => {
+                await assertCommandError(commandAssertionDependencies, {
+                    systemUserId: dummySystemUserId,
+                    buildCommandFSA: buildValidCommandFSA,
+                    initialState: new DeluxeInMemoryStore({
```

Comment:
We should add a `checkErrors` property here. Make sure we get the corresponding `CannotAddDuplicateItemToPlaylist` error.